synapse/rest/client/v1/login.py

```diff
@@ -390,7 +390,9 @@ def __init__(self, hs):
     async def get_sso_url(
         self, request: SynapseRequest, client_redirect_url: bytes
     ) -> bytes:
-        return self._saml_handler.handle_redirect_request(client_redirect_url)
+        return await self._saml_handler.handle_redirect_request(
+            request, client_redirect_url
+        ).encode("utf8")
```

Comment:
we're encoding something we've just decoded, sadly. Still it's a win for consistency's sake.